packages/scxa-marker-gene-heatmap/src/MarkerGeneHeatmap.js

```diff
@@ -59,60 +59,56 @@ Highcharts.SVGRenderer.prototype.symbols.download = (x, y, w, h) => [
-  const dynamicHeight = (totalNumberOfRows * heatmapRowHeight) + (cellTypes.length * 8) + 175
+  const dynamicHeight = (totalNumberOfRows * heatmapRowHeight) + (xAxisCategories.length * 8) + 175
 
+  const groupedData = _.groupBy(data, `cellGroupValueWhereMarker`)
   let plotLineAxisPosition = -0.5
 
-  // If we don't have a set row height, we try to estimate the height as worked out by Highcharts
-  const rowHeight = hasDynamicHeight ?
-    heatmapRowHeight :
-    Math.round((chartHeight - 175) / totalNumberOfRows + ((cellTypes.length - 1) * 8))
-
-  cellTypes.forEach((cellType, idx, array) => {
-    const numberOfRows = Object.keys(_.groupBy(groupedData[cellType], `y`)).length // how many marker genes per cluster
-    plotLineAxisPosition = plotLineAxisPosition + numberOfRows
-
-    const yOffset = -numberOfRows * rowHeight / 2
-
-    let color, zIndex
-    // don't show last plot line
-    if (idx === array.length - 1) {
-      // removing the plot line altogether would remove the label, so we need to make it "invisible"
-      color = `#FFFFFF`
-      zIndex = 0
-    } else {
-      color = `#000000`
-      zIndex = 5
+  cellTypesOrClusterIds.length === 1 ? plotLines.push({
+    color: `#000000`,
+    width: 2,
+    value: data[0].y,
+    zIndex: 0,
+    label: heatmapType !== `multiexperimentcelltypes` && {
+      text: splitPhrase(heatmapOptionsProvider[heatmapType].labelsFormatter(cellTypesOrClusterIds[0])).join(`<br/>`),
+      align: `right`,
+      textAlign: `left`,
+      x: 0,
+      y: 30
     }
-
-    const splitCellTypeLabel = splitPhrase(heatmapOptionsProvider[heatmapType].labelsFormatter(cellType))
-    plotLines.push({
-      color: color,
-      width: 2,
-      value: plotLineAxisPosition,
-      zIndex: zIndex,
-      label: heatmapType !== `multiexperimentcelltypes` && {
-        useHTML: true,
-        text: splitCellTypeLabel.join(`<br/>`),
-        align: `right`,
-        textAlign: `left`,
-        x: 15,
-        y: yOffset - Math.max(splitCellTypeLabel.length - 3, 0) * rowHeight / 2 // Move very long labels slightly up
-      }
+  }) :
+    cellTypesOrClusterIds.map((cellTypeOrClusterId) => {
+      const numberOfRows = Object.keys(_.groupBy(groupedData[cellTypeOrClusterId.toString()], `y`)).length // how many marker genes per cluster
+      plotLineAxisPosition = plotLineAxisPosition + numberOfRows
+      const yOffset = -numberOfRows * heatmapRowHeight / 2
+      const color = `#000000`
+      const zIndex = 5
+      const splitCellTypeLabel = splitPhrase(heatmapOptionsProvider[heatmapType].labelsFormatter(cellTypeOrClusterId))
+      plotLines.push({
+        color,
+        width: 2,
+        value: plotLineAxisPosition,
+        zIndex,
+        label: heatmapType !== `multiexperimentcelltypes` && {
+          text: splitCellTypeLabel.join(`<br/>`),
+          align: `right`,
+          textAlign: `left`,
+          x: 0,
```

Comment:
Originally we added 15 pixels of separation between the heatmap and the labels. Could we have it back? You may need to change the options of the plot to reserve more space for the labels in that case.

Reply:
This is the bug fix to have the label back, so we can't reset it back to `15`.

Reply:
It turns out that there is a bug in Highcharts. I’ve reported it here:
https://github.com/highcharts/highcharts/issues/18149

In the meantime I’ve got a fix planned to add a bit of space to the label, but I’ll open a separate PR for that. :)

Reply:
It was a duplicate of https://github.com/highcharts/highcharts/issues/17888 and there are two workarounds, which we’ll add once this is merged.